components/core/src/LogTypeDictionaryWriter.cpp

```diff
@@ -47,20 +38,18 @@ void LogTypeDictionaryWriter::open_and_preload (const std::string& dictionary_pa
     m_is_open = true;
 }
 
-bool LogTypeDictionaryWriter::add_occurrence (std::unique_ptr<LogTypeDictionaryEntry>& entry_wrapper, logtype_dictionary_id_t& logtype_id) {
-    if (nullptr == entry_wrapper) {
+bool LogTypeDictionaryWriter::add_occurrence (LogTypeDictionaryEntry& logtype_entry, logtype_dictionary_id_t& logtype_id) {
+    if (logtype_entry.get_value().empty()) {
         throw OperationFailed(ErrorCode_BadParam, __FILENAME__, __LINE__);
     }
-    auto& entry = *entry_wrapper;
 
     bool is_new_entry = false;
 
-    const string& value = entry.get_value();
-    const auto ix = m_value_to_entry.find(value);
-    if (m_value_to_entry.end() != ix) {
-        // Entry exists so increment its count
-        auto& existing_entry = ix->second;
-        logtype_id = existing_entry->get_id();
+    const string& value = logtype_entry.get_value();
```

Comment:
shouldn't it be moved to above line 16? we should verifiy that the log entry string is not empty for both look up and insertion.

Reply:
Uh, I'm not sure we're referring to the same line numbers any more. What I meant was you could put the `const string& value = logtype_entry.get_value();` above `if (logtype_entry.get_value().empty())` so that you could instead write `if (value.empty())`